tests/prague/eip7692_eof_v1/eip3540_eof_v1/test_all_opcodes_in_container.py

```diff
@@ -129,3 +131,271 @@ def test_all_opcodes_in_container(
+    ],
+)
+def test_all_invalid_terminating_opcodes(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test all opcodes that are invalid as the last opcode in a container
+    """
+    if opcode.has_data_portion():
+        opcode_with_data_portion = opcode[0] if opcode != Op.CALLF else opcode[1]
+    else:
+        opcode_with_data_portion = opcode
+
+    bytecode = (Op.ORIGIN * opcode_with_data_portion.min_stack_height) + opcode_with_data_portion
+
+    sections = [Section.Code(code=bytecode)]
+
+    if opcode == Op.CALLF:
+        sections += [Section.Code(code=Op.RETF, code_outputs=0)]
+    elif opcode == Op.EOFCREATE:
+        sections += [
+            Section.Container(
+                container=Container(
+                    sections=[
+                        Section.Code(code=Op.RETURNCONTRACT[0](0, 0)),
+                        Section.Container(Container.Code(code=Op.STOP)),
+                    ]
+                )
+            )
+        ]
+
+    sections += [Section.Data(b"\0" * 32)]
+
+    eof_test(
+        data=Container(
+            sections=sections,
+        ),
+        expect_exception=EOFException.MISSING_STOP_OPCODE,
+    )
+
+
+@pytest.mark.parametrize(
+    "opcode",
+    [op for op in list(Op) if op in halting_opcodes],
+)
+def test_all_unreachable_terminating_opcodes_after_stop(
+    eof_test: EOFTestFiller,
+    opcode: Opcode,
+):
+    """
+    Test all opcodes terminating opcodes after stop.
+    """
+    match opcode:
+        case Op.STOP:
+            sections = [Section.Code(code=Op.STOP + Op.STOP)]
+        case Op.RETF:
+            sections = [
+                Section.Code(code=Op.STOP + Op.CALLF[1]),
+                Section.Code(code=Op.STOP + Op.RETF, code_outputs=0),
+            ]
+        case Op.JUMPF:
+            sections = [
+                Section.Code(code=Op.STOP + Op.JUMPF[1]),
+                Section.Code(code=Op.STOP),
+            ]
+        case Op.RETURNCONTRACT:
+            sections = [
+                Section.Code(code=Op.EOFCREATE[0](0, 0, 0, 0) + Op.STOP),
+                Section.Container(
+                    container=Container(
+                        sections=[
+                            Section.Code(code=Op.STOP + Op.RETURNCONTRACT[0](0, 0)),
+                            Section.Container(Container.Code(code=Op.STOP)),
+                        ]
+                    )
+                ),
+            ]
+        case Op.RETURN | Op.REVERT | Op.INVALID:
+            sections = [
+                Section.Code(code=Op.PUSH1(0) + Op.PUSH1(0) + Op.STOP + opcode),
+            ]
+        case _:
+            raise NotImplementedError(f"Opcode {opcode} is not implemented")
+
+    eof_test(
+        data=Container(
+            sections=sections,
+        ),
+        expect_exception=EOFException.UNREACHABLE_INSTRUCTIONS
+        if opcode != Op.RETURNCONTRACT
+        else EOFException.INCOMPATIBLE_CONTAINER_KIND,
```

Comment:
I'm not sure there should be a special case for the `RETURNCONTRACT`.

Reply:
It seems to me like the subcontainer that should be initcode it's being detected as runtime container because the `STOP` is found first before the `RETURNCONTRACT`.

Reply:
ok, I get it. I think the `REVERT` instead of `STOP` may prevent it, but I'm not sure this qualifies as the same test.